renderer.go

```diff
@@ -0,0 +1,144 @@
+package lipgloss
+
+import (
+	"github.com/muesli/termenv"
+)
+
+var renderer = NewRenderer()
+
+// Renderer is a lipgloss terminal renderer.
+type Renderer struct {
+	output            *termenv.Output
+	hasDarkBackground *bool
+}
+
+// RendererOption is a function that can be used to configure a Renderer.
+type RendererOption func(r *Renderer)
+
+// DefaultRenderer returns the default renderer.
+func DefaultRenderer() *Renderer {
+	return renderer
+}
+
+// NewRenderer creates a new Renderer.
+func NewRenderer(options ...func(r *Renderer)) *Renderer {
+	r := &Renderer{
+		output: termenv.DefaultOutput(),
+	}
+	for _, option := range options {
+		option(r)
+	}
+	return r
+}
+
+// WithOutput sets the termenv Output to use for rendering.
+func WithOutput(output *termenv.Output) RendererOption {
```

Comment:
Similar to what we plan in `tea`, we should probably have a `WithOutput` that expects a regular file descriptor and rename this implementation to `WithTermenvOutput`.